x/adminmodule/keeper/proposal.go

```diff
@@ -7,39 +7,38 @@ import (
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	govv1types "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
-
-	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govv1types "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 )
 
 // SubmitProposal create new proposal given a content
-func (k Keeper) SubmitProposal(ctx sdk.Context, content govv1types.Content) (govv1types.Proposal, error) {
-	if !k.rtr.HasRoute(content.ProposalRoute()) {
-		return govv1types.Proposal{}, sdkerrors.Wrap(govtypes.ErrNoProposalHandlerExists, content.ProposalRoute())
-	}
-
-	cacheCtx, _ := ctx.CacheContext()
-	handler := k.rtr.GetRoute(content.ProposalRoute())
-	if err := handler(cacheCtx, content); err != nil {
-		return govv1types.Proposal{}, sdkerrors.Wrap(govtypes.ErrInvalidProposalContent, err.Error())
-	}
-
+func (k Keeper) SubmitProposal(ctx sdk.Context, msgs []sdk.Msg) (govv1types.Proposal, error) {
```

Comment:
I'm proposing to create another task not to burned Pasha, but we need to rename SubmitProposal() -> ExecuteProposal() and SubmitLegacyProposal() -> ExecuteLegacyProposal()
Because now it's very misleading to read